tests/libraries/dynamic_ref/test.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+set -o pipefail
+
+rlJournalStart
+    rlPhaseStartSetup "Prepare git repos"
+        testdir=$PWD
+        rlRun "repo1=\$(mktemp -d)"
+        rlRun "repo2=\$(mktemp -d)"
+        # create test repo
+        rlRun "cp data/test.sh $repo1"
+        cat <<EOF > $repo1/main.fmf
+/first:
+  test: bash test.sh
+  framework: beakerlib
+  require:
+  - name: /dynref
+    type: library
+    url: $repo2
+    ref: "@dynamic-ref"
+EOF
+        rlRun "tmt init $repo1"
+        rlRun "pushd $repo1"
+        rlRun "git init"
+        rlRun "git config --local user.email me@localhost.localdomain"
+        rlRun "git config --local user.name m e"
+        rlRun "git add -A"
+        rlRun "git commit -m initial"
+        rlRun "popd"
+        # create library repo
+        rlRun "tmt init $repo2"
+        rlRun "pushd $repo2"
+        rlRun "git init"
+        rlRun "git config --local user.email me@localhost.localdomain"
+        rlRun "git config --local user.name m e"
+        rlRun "cp $testdir/data/dynamic-ref $repo2"
+        rlRun "git add -A"
+        rlRun "git commit -m first"
+        rlRun "git checkout -b otherbranch"
+        rlRun "mkdir dynref"
+        rlRun "cp -r $testdir/data/lib.sh $repo2/dynref"
+        cat <<EOF > $repo2/dynref/main.fmf
+test: bash lib.sh
+framework: beakerlib
+EOF
+        rlRun "git add -A"
+        rlRun "git commit -m second"
+        rlRun "git checkout master"
```

Comment:
In the test log there's the default branch name warning:

```
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint: 
hint: 	git config --global init.defaultBranch <name>
hint: 
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint: 
hint: 	git branch -m <name>
```

Let's set the `init.defaultBranch` as recommended to prevent future failure when the default changes.

Reply:
Ack

Reply:
Fixed in c376cf1c.